main.go

```diff
@@ -153,18 +158,17 @@ func (srv *Server) init() error {
 // our counter (i.e., an integer overflow is about to happen), we return an
 // error, which signals to the caller that it's time to create a new randomness
 // server instance.
-func (srv *Server) puncture() error {
+func (srv *Server) puncture(md epoch) error {
 	srv.Lock()
 	defer srv.Unlock()
 
-	C.randomness_server_puncture(srv.raw, (C.uint8_t)(srv.md))
+	C.randomness_server_puncture(srv.raw, (C.uint8_t)(md))
 
 	// An epoch is exhausted when our 8-bit counter is about to overflow.
-	if srv.md == maxEpoch {
+	if md == maxEpoch {
```

Comment:
It looks like we still need to signal to the caller that all epochs have been exhausted, to trigger a reinitialization of the server instance. Correct me if I'm wrong

Reply:
Yes, you are right. I think we're better off moving this check into `epochLoop` and have `puncture` do nothing but calling the FFI function. What do you think?

Reply:
If the plan is to eventually replace `epochLoop` (as discussed here: https://github.com/brave-experiments/star-randsrv/pull/10#discussion_r902915834), I think it would be better to hold off. I don't think the current approach is a bad one.

Reply:
That's fair.